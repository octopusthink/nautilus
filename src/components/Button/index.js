import { css } from '@emotion/core';
import invariant from 'invariant';
import PropTypes from 'prop-types';
import React from 'react';

export const Button = ({
  children,
  minimal,
  navigation,
  primary,
  ...otherProps
}) => {
  invariant(
    (minimal === true && primary === true) === false,
    'Cannot use `minimal` and `primary` props together on a <Button> component.',
  );

  let Component = 'button';
  if (navigation === true) {
    Component = 'a';
  }

  return (
    // See: https://github.com/yannickcr/eslint-plugin-react/issues/1555
    // eslint-disable-next-line react/button-has-type
    <Component
      css={css`
        font-family: -apple-system;
        font-size: 17px;
        padding: 12px 32px;
        outline: none;
        text-decoration: none;
        border: 2px solid #c62e7f;
        color: #c62e7f;
        background: #fff;
        border-radius: 8px;
        margin: 0 16px 16px;
        line-height: 1.412;
        transition: box-shadow 200ms;

        &:active {
          border-color: #973367;
          color: #973367;
        }

        &:focus {
          box-shadow: 0 0 1px 4px #F5A3CE;
          outline: none;
        }

        &::-moz-focus-inner {
          border: 0;
        }

        &:hover {
          box-shadow: 0 2px 0 0 #973367;
        }

        ${primary &&
          css`
            background: #c62e7f;
            color: #fff;

            &:active {
              background: #973367;
              color: #fff;
            }
          `}

        ${minimal &&
          css`
            border-top: 0;
            border-left: 0;
            border-right: 0;
            border-radius: 0;
            padding-left: 0;
            padding-right: 0;
          `}

        ${navigation &&
          css`
            &::after {
              content: ' â†’';
              display: inline;
            }
          `}

          &:disabled {
            border-color: #9f9f9f;
            color: #9f9f9f;

            &:hover {
                box-shadow: 0 2px 0 0 #6C6C6C;
            }

            &:active {
              background: #6C6C6C;
            }

            &:focus {
              box-shadow: 0 0 1px 4px #D4D4D4;
            }

            &${primary &&
              css`
                background: #9f9f9f;
                color: #fff;
              `}

      `}
      {...otherProps}
    >
      {children}
    </Component>
  );
};

Button.defaultProps = {
  children: undefined,
  disabled: false,
  minimal: false,
  navigation: false,
  primary: false,
  type: 'button',
};

Button.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Disables the button; this applies a disabled style but **does not disable any event handlers for the button**. Your `onClick` handlers should check for `props.disabled` to modify their behaviour accordingly. */
  disabled: PropTypes.bool,
  /** Decrease the visual prominence of this button in the UI. */
  minimal: PropTypes.bool,
  /** Sometimes, you need a button that's actually a link. We got you! This outputs a `react-router-dom` `<Link>` tag that looks (and largely behaves) like a `<Button>`, but can used as navigation. Setting this to `true` enables `Link` properties; see: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md. */
  navigation: PropTypes.bool,
  /** Increase the visual prominence of this button in the UI. */
  primary: PropTypes.bool,
  /** HTML `type` attribute for the button. Defaults to `"button"`. */
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
};

export default Button;
