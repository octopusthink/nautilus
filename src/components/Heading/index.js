import { css } from '@emotion/core';
import invariant from 'invariant';
import PropTypes from 'prop-types';
import React from 'react';

import { useTheme } from 'themes';
import { headingLarge, headingMedium, headingSmall } from 'themes/mixins';

export const Heading = ({
  children,
  level,
  large,
  medium,
  small,
  ...otherProps
}) => {
  invariant(level, 'Heading component requires a level prop');

  const HeadingElement = `h${level}`;
  const theme = useTheme();

  return (
    <HeadingElement
      css={css`
        ${css(headingMedium(theme))};
        margin: 0 0 ${theme.spacing.margin.m};

        ${large &&
          css`
            ${css(headingLarge(theme))}
          `}

        ${small &&
          css`
            ${css(headingSmall(theme))}
          `}
      `}
      {...otherProps}
    >
      {children}
    </HeadingElement>
  );
};

export const HeadingLevels = [1, 2, 3, 4, 5, 6];

Heading.defaultProps = {
  children: undefined,
  level: 2,
  large: true,
  medium: false,
  small: false,
};

Heading.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Semantic hierarchy level of the `<h>` element in the markup (ex: `<h3>`) */
  level: PropTypes.oneOf(HeadingLevels).isRequired,
  /** Visual prominency of the heading. Small, medium, large. */
  large: PropTypes.bool,
  medium: PropTypes.bool,
  small: PropTypes.boolm,
};

export default Heading;
