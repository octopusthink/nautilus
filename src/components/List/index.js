import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { bodySmall, bodyMedium, bodyLarge } from 'themes/mixins';

export const List = ({
  children,
  large,
  small,
  inverse,
  dark,
  light,
  ...otherProps
}) => {
  return <ul {...otherProps}>{children}</ul>;
};

List.defaultProps = {
  children: undefined,
  large: false,
  small: false,
  inverse: false,
  dark: false,
  light: false,
};

List.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Increase the visual prominence of the paragraph. */
  large: PropTypes.bool,
  /** Decrease the visual prominence of the paragraph. */
  small: PropTypes.bool,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** Lighten text colour. */
  light: PropTypes.bool,
};

export default styled(List)(({ dark, inverse, large, light, small, theme }) => {
  return css`
      color: ${theme.colors.text.default};
      margin: 0 0 ${theme.spacing.margin.m};
      ${small && bodySmall(theme)};
      ${!small && !large && bodyMedium(theme)};
      ${large && bodyLarge(theme)};
      ${light &&
        css`
          color: ${theme.colors.text.light};
        `}

      ${dark &&
        css`
          color: ${theme.colors.text.dark};
        `}

      ${inverse &&
        css`
          color: ${theme.colors.text.inverse};

          ${light &&
            css`
              color: ${theme.colors.text.inverseLight};
            `}
          ${dark &&
            css`
              color: ${theme.colors.text.inverseDark};
            `}
        `}
    `;
});
