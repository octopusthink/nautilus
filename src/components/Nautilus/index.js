import { Global, css } from '@emotion/core';
import { ThemeProvider } from 'emotion-theming';
import PropTypes from 'prop-types';
import React from 'react';

import { nautilus as nautilusDefaultTheme } from 'themes';

export const Nautilus = ({ children, theme }) => {
  return (
    <ThemeProvider theme={theme}>
      <Global
        styles={css`
          html {
            /* Makes type rendering look crisper. */
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;

            /* Set base font size for rem units to inherit .*/
            font-size: 62.5%;

            /* Enable kerning and optional ligatures. */
            text-rendering: optimizeLegibility;

            /* Disable auto-enlargement of small text in Safari. */
            text-size-adjust: 100%;
          }
        `}
      />
      {children}
    </ThemeProvider>
  );
};

Nautilus.defaultProps = {
  children: undefined,
  theme: nautilusDefaultTheme,
};

Nautilus.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Theme object used to style this instance of Nautilus and its components. */
  theme: PropTypes.object,
};

export default Nautilus;
