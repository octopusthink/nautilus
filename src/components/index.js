export { default as Heading } from './Heading';
export { default as Paragraph } from './Paragraph';
