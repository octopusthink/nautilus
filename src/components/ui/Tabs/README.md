Tabs allows users to quickly navigate between related views in the same context and at the same level of hierarchy. Tabs display one section at a time.

Individual tabs (`Tabs.Tab`) are contained within a `Tabs` component:

```jsx
import { Paragraph } from '@octopusthink/nautilus';

<Tabs>
  <Tabs.Tab label="About">
    <Paragraph>Puppies are cute.</Paragraph>
	</Tabs.Tab>

	<Tabs.Tab label="History">
    <Paragraph>Puppies have already been cute.</Paragraph>
	</Tabs.Tab>

	<Tabs.Tab label="Learn more">
    <Paragraph>Have you seen the <a href="https://www.vox.com/culture/2017/2/4/14480722/puppy-bowl-explained">Puppy Bowl</a>?</Paragraph>
	</Tabs.Tab>
</Tabs>
```

## Usage

A tab should be used to represent content that is related and of equal weight or importance, with the caveat that the first section is the most relevant to most users.

### Use this component for...

Tabs can be a helpful way of letting users quickly switch between related information if:

- Your content can be usefully separated into clearly labelled sections.
- Users will not need to view all the sections at once.

### Don't use this component for...

- Primary navigation for your site. Use a Menu instead.
- Representing content that is dissimilar. Separate this into different pages, or use Headings to section off your page.
- Content that should be read in order, or that requires a comparison across different tabs. Anything that requires users to jump back and forth between Tabs will be very frustrating to use.
- A group of more than six items. Use different pages or sections instead.

## Appearance

### Selected tabs

The currently-selected tab inherits some styling to aid users in understanding where they are in the current tab structure.

Only one Tab may be active at a time.

When the page loads, the first Tab in a sequence will always be selected.

### Alignment

A TabGroup can either be aligned to the top, or to the side. Alignment of tabs will follow the logical reading order of text (RTL or LTR, depending on languages.

### Tab width

The width of a Tab will adjust to the length of its label. If the list of Tabs cannot fit in the space available, it will automatically be displayed in a scrollable area with next/previous arrow buttons to navigate.

Test your designs on mobile devices and other small screens to see if this could potentially cause issues.

## Behaviour

### One tab is active at a time

Only one tab should be active at a time.

### Number of tabs

Do not use more than six tabs together. Using more than this can lead to a cluttered UI and needless cognitive overload for users. Instead, break long groups up into more relevant sections.

### Ordering tabs

Order tabs in the most logical progression. The first tab in a group will be visible by default, so the most important content should be here. Arrange the other tabs in the order that makes most sense for your users.

### Don't disable tabs

Disabling elements is normally confusing for users. If there is no content for a tab, either remove the tab or, if that would be confusing for your users, explain why there is no content when the tab is selected. To this point: **Nautilus does not offer an API to disable tabs.**

## Voice & tone

- Use sentence case for tab labels.
- Use clear labels to help differentiate the different sections beneath them. If you struggle to come up with clear labels, it might be because the way youâ€™ve separated the content is not clear.
- Use short, easily-scannable labels. A single word is best. Do not exceed more than three words per tab label.

## Accessibility

- Don't disable tabs or create tabs that don't link to content. This is confusing to users; instead either remove the tab from the UI or offer an alternative explanation inside the tab's content as to why its content isn't available.
- Give tabs clear, distinct labels via the `label` prop. These labels are used by assistive technologies to describe each tab to the user, so they should be text-only.
- Tabs should not be used as navigation. Selecting a tab should change the content displayed within a tab group but should keep the user on the existing page.
- Question whether you actually need to be using a tabbed interface when sections of content, marked by a Heading component, would work.

### Don't hide important content

Because some content will always be hidden in a Tab, do not use Tabs for any information that is critical to the user. Use Tabs to help users find the content they need more easily. Tabs are helpful for users who wish to switch between different sections quickly without having to wade back up the page each time they want to choose a new one.

### Keyboard navigation

Users can switch between Tabs by using the arrow keys. This allows for easier backward- and forward- navigation than using the tab key, which would force users to cycle through all Tabs to go back to the last one. The tab key is used to focus on a Tab and displaying its contents.

Pressing `shift` + `tab` returns the user to the selected tab.

All unselected Tabs are assigned `tabindex="-1"`, which removes the inactive tabs from focus order but allows focus via a script.

### Aria settings

- The `Tabs` component has a role of `tablist`. The `Tabs.Tab` label has a role of `tab`. The `Tabs.Tab` content has a role of `tabpanel`.
- The currently selected tab has `aria-selected` set to `true`. All other tabs set this value to `false`.
- Tabs has `aria-orientation` set to `vertical` or `horizontal` based on the value of the `orientation` option.
- Each tab has `aria-controls` referring to its associated tab content.

### Labels label the Tab content

The Tab content uses `aria-labelledby` to tie it to its associated `Tab`.

Since tab panels are labeled by their tabs in this way, when the down arrow is pressed and the relevant tab panel focused, a screen reader will announce: `"[tab label], tab panel"`, thereby assuring the user of their new location within the interface.
