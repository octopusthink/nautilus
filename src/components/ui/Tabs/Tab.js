import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { toUnits } from 'styles';
import { useTheme } from 'themes';

export const Tab = (props) => {
  const { children, label, ...otherProps } = props;
  const theme = useTheme();

  return (
    <li role="presentation">
      <a
        css={css`
          padding: ${toUnits(theme.spacing.padding.small)};
          text-decoration: none;
          color: ${theme.colors.neutral.grey800};
        `}
        role="tab"
        href="#section1"
        id="tab1"
        aria-selected="true"
      >
        {label}
      </a>
    </li>
  );
};

export const styles = (props) => {
  const { theme } = props;

  return css`
    margin: 0 ${toUnits(theme.spacing.padding.small)} 0 0;
  `;
};

Tab.defaultProps = {
  children: undefined,
};

Tab.propTypes = {
  /** @ignore */
  children: PropTypes.node,
};

Tab.displayName = 'Tabs.Tab';

export default styled(Tab)(styles);
