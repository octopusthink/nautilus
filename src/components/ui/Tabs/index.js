import { css } from '@emotion/react';
import PropTypes from 'prop-types';
import React, {
  Children,
  cloneElement,
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from 'react';
import shortid from 'shortid';
import { useUpdateEffect } from 'react-use';
import { focusStyle, bodyStyles, toUnits } from '../../../styles';
import { useTheme } from '../../../themes';
import Tab from './Tab';

const isTab = (child) => child.type === Tab;

const setFocus = (focusedElement, ref, setStateFunc) => {
  if (focusedElement !== null && ref && ref.current) {
    ref.current.focus();
    setStateFunc(null);
  }
};

const Tabs = (props) => {
  const {
    activeTab: activeTabProp,
    children,
    dark,
    inverse,
    light,
    noMargin,
    id,
    onTabChange,
    unstyled,
    ...otherProps
  } = props;
  const sectionToFocusRef = useRef();
  const tabToFocusRef = useRef();
  const [generatedId] = useState(shortid.generate());
  const [activeTabState, setActiveTab] = useState(0);
  const [focusedSection, setFocusedSection] = useState(null);
  const [focusedTab, setFocusedTab] = useState(null);
  const theme = useTheme();

  // Allow the active tab to be managed by a prop.
  const activeTab = activeTabProp ?? activeTabState;

  const numberOfTabs = useMemo(() => Children.toArray(children).filter(isTab).length, [children]);

  // Assigns focus to a tab's content when the down arrow key is pressed.
  useEffect(() => {
    setFocus(focusedSection, sectionToFocusRef, setFocusedSection);
  }, [focusedSection, sectionToFocusRef]);

  // Assign focus to a tab's label when the tab changes via the left/right keys.
  useEffect(() => {
    setFocus(focusedTab, tabToFocusRef, setFocusedTab);
  }, [focusedTab, tabToFocusRef]);

  const onKeyDown = useCallback(
    (event) => {
      let newTabIndex;

      // 37 is the keycode for left arrow, which will activate the previous tab.
      if (event.which === 37) {
        newTabIndex = activeTab - 1;
      }
      // 39 is the keycode for right arrow, which will activate the next tab.
      if (event.which === 39) {
        newTabIndex = activeTab + 1;
      }

      // If a new tab index was set, prevent the default action, advance to
      // that tab, and then return;
      if (newTabIndex !== undefined && newTabIndex >= 0 && newTabIndex < numberOfTabs) {
        event.preventDefault();
        setActiveTab(newTabIndex);
        setFocusedTab(newTabIndex);
        return;
      }

      // If these key presses weren't able to move the focused tab because
      // we're at the end of the row, at least ensure the correct tab is
      // still focused.
      if (newTabIndex !== undefined) {
        event.preventDefault();
        setFocusedTab(activeTab);
        return;
      }

      // 40 is the keycode for down arrow, which will move the focus into the
      // active tab's content.
      if (event.which === 40) {
        event.preventDefault();
        setFocusedSection(activeTab);
      }
    },
    [activeTab, numberOfTabs],
  );

  const onClickFactory = (index) => (event) => {
    event.preventDefault();
    setActiveTab(index);
  };

  useUpdateEffect(() => {
    if (onTabChange) {
      onTabChange(activeTabState);
    }
  }, [activeTabState]);

  const tabsId = id || generatedId;

  const labels = useMemo(
    () =>
      Children.toArray(children)
        .filter(isTab)
        .map((child, index) => {
          const { label, labelTestId, ...labelProps } = child.props;
          const {
            onClick: onClickTab,
            onKeyDown: onKeyDownTab,
            ref: refTab,
            ...otherTabProps
          } = labelProps;

          return (
            <li key={`${tabsId}-tab-${label}`} role="presentation">
              <a
                aria-selected={index === activeTab}
                css={css`
                  color: ${theme.colors.text.default};
                  display: inline-block;
                  padding: ${toUnits(theme.spacing.padding.small)}
                    ${toUnits(theme.spacing.padding.large)};
                  position: relative;
                  text-decoration: none;

                  &::after {
                    display: block;
                    border-bottom: 3px solid transparent;
                    content: '';
                    position: absolute;
                    bottom: -2px;
                    left: 0;
                    right: 0;
                  }

                  ${index === activeTab &&
                  css`
                    color: ${theme.colors.text.dark};

                    &::after {
                      border-bottom: 3px solid;
                    }
                  `}

                  &:hover {
                    /* color: ${theme.colors.accent.primaryDark}; */

                    &::after {
                      border-color: ${theme.colors.neutral.grey400};
                    }
                  }

                  &:focus {
                    ${focusStyle.outline(theme)};
                  }
                `}
                href={`#${tabsId}-section-${index}`}
                id={`${tabsId}-tab-${index}`}
                onClick={(event) => {
                  onClickFactory(index)(event);
                  if (onClickTab) {
                    onClickTab(event, index);
                  }
                }}
                onKeyDown={(event) => {
                  onKeyDown(event);
                  if (onKeyDownTab) {
                    onKeyDownTab(event, index);
                  }
                }}
                ref={focusedTab === index ? tabToFocusRef : refTab}
                role="tab"
                tabIndex={index !== activeTab ? '-1' : undefined}
                {...otherTabProps}
                data-testid={labelTestId}
              >
                {label}
              </a>
            </li>
          );
        }),
    [children, activeTab, focusedTab, tabsId, onKeyDown, tabToFocusRef, theme],
  );

  const tabPanels = useMemo(
    () =>
      Children.toArray(children)
        .filter(isTab)
        .map((child, index) => {
          const { label } = child.props;

          return cloneElement(child, {
            'aria-labelledby': `${tabsId}-tab-${index}`,
            isActive: activeTab === index,
            id: `${tabsId}-section-${index}`,
            key: `${tabsId}-section-${label}`,
            ref: focusedSection === index ? sectionToFocusRef : undefined,
          });
        }),
    [children, activeTab, focusedSection, tabsId, sectionToFocusRef],
  );

  return (
    <React.Fragment>
      {labels && !!labels.length && (
        <ul
          css={css`
            ${bodyStyles({ dark, inverse, light, noMargin, theme })};
            border-bottom: 1px solid ${theme.colors.neutral.grey600};
            display: flex;
            list-style-type: none;
            padding: 0;
          `}
          role="tablist"
          {...otherProps}
        >
          {labels}
        </ul>
      )}
      {tabPanels}
    </React.Fragment>
  );
};

Tabs.defaultProps = {
  children: undefined,
  dark: false,
  id: undefined,
  inverse: false,
  light: false,
  noMargin: false,
  unstyled: false,
};

Tabs.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** @ignore HTML ID prop */
  id: PropTypes.string,
  /** Lighten text colour. */
  light: PropTypes.bool,
  /** Remove any outer margins from component. */
  noMargin: PropTypes.bool,
  /* @ignore Don't output any CSS styles. */
  unstyled: PropTypes.bool,
};

export const { defaultProps, propTypes } = Tabs;

// Export Tab as `Tabs.Tab`.
Tabs.Tab = Tab;

export default Tabs;
