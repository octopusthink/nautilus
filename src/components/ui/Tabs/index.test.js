import React from 'react';

import { muteConsole, render } from 'utils/testing';

import { Heading, Paragraph } from 'components';
import Tabs from '.';

describe('Tabs', () => {
  it('should render a <ul> and a <section>', () => {
    const { container } = render(<Tabs />);

    expect(container.firstChild.tagName).toEqual('UL');
  });
});
