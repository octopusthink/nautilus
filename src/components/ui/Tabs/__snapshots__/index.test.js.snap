// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should not render styles when unstyled prop is set 1`] = `
.emotion-0 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
  border-bottom: 1px solid #97a0af;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  padding: 0;
}

.emotion-1 {
  color: #3b3f45;
  display: inline-block;
  padding: 0.8rem 2.4rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #181b1c;
}

.emotion-1::after {
  display: block;
  border-bottom: 3px solid transparent;
  content: '';
  position: absolute;
  bottom: -2px;
  left: 0;
  right: 0;
}

.emotion-1:hover::after {
  border-color: #bec4cd;
}

.emotion-1:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-1::after {
  border-bottom: 3px solid;
}

.emotion-2:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-3 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

<div>
  <ul
    class="emotion-0"
    role="tablist"
  >
    <li
      role="presentation"
    >
      <a
        aria-selected="true"
        class="emotion-1"
        data-testid="firstTab"
        href="#myTabSet-section-0"
        id="myTabSet-tab-0"
        role="tab"
      >
        About
      </a>
    </li>
  </ul>
  <section
    aria-labelledby="myTabSet-tab-0"
    class="emotion-2"
    data-testid="firstTabSection"
    id="myTabSet-section-0"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-3"
    >
      Puppies are cute.
    </p>
  </section>
</div>
`;

exports[`Tabs should render styles for Tabs 1`] = `
.emotion-0 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
  border-bottom: 1px solid #97a0af;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  padding: 0;
}

.emotion-1 {
  color: #3b3f45;
  display: inline-block;
  padding: 0.8rem 2.4rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #181b1c;
}

.emotion-1::after {
  display: block;
  border-bottom: 3px solid transparent;
  content: '';
  position: absolute;
  bottom: -2px;
  left: 0;
  right: 0;
}

.emotion-1:hover::after {
  border-color: #bec4cd;
}

.emotion-1:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-1::after {
  border-bottom: 3px solid;
}

.emotion-2 {
  color: #3b3f45;
  display: inline-block;
  padding: 0.8rem 2.4rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2::after {
  display: block;
  border-bottom: 3px solid transparent;
  content: '';
  position: absolute;
  bottom: -2px;
  left: 0;
  right: 0;
}

.emotion-2:hover::after {
  border-color: #bec4cd;
}

.emotion-2:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-4:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-5 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

.emotion-6 {
  display: none;
}

.emotion-6:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

<div>
  <ul
    class="emotion-0"
    role="tablist"
  >
    <li
      role="presentation"
    >
      <a
        aria-selected="true"
        class="emotion-1"
        data-testid="firstTab"
        href="#myTabSet-section-0"
        id="myTabSet-tab-0"
        role="tab"
      >
        About
      </a>
    </li>
    <li
      role="presentation"
    >
      <a
        aria-selected="false"
        class="emotion-2"
        data-testid="secondTab"
        href="#myTabSet-section-1"
        id="myTabSet-tab-1"
        role="tab"
        tabindex="-1"
      >
        History
      </a>
    </li>
    <li
      role="presentation"
    >
      <a
        aria-selected="false"
        class="emotion-2"
        data-testid="thirdTab"
        href="#myTabSet-section-2"
        id="myTabSet-tab-2"
        role="tab"
        tabindex="-1"
      >
        Other
      </a>
    </li>
  </ul>
  <section
    aria-labelledby="myTabSet-tab-0"
    class="emotion-4"
    data-testid="firstTabSection"
    id="myTabSet-section-0"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-5"
    >
      Puppies are cute.
    </p>
  </section>
  <section
    aria-labelledby="myTabSet-tab-1"
    class="emotion-6"
    data-testid="secondTabSection"
    id="myTabSet-section-1"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-5"
    >
      Puppies have already been cute.
    </p>
  </section>
  <section
    aria-labelledby="myTabSet-tab-2"
    class="emotion-6"
    data-testid="thirdTabSection"
    id="myTabSet-section-2"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-5"
    >
      I have run out of things to say in this test.
    </p>
  </section>
</div>
`;

exports[`Tabs shouldn't have a margin when \`noMargin\` is true 1`] = `
.emotion-0 {
  color: #3b3f45;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
  border-bottom: 1px solid #97a0af;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style-type: none;
  padding: 0;
}

.emotion-1 {
  color: #3b3f45;
  display: inline-block;
  padding: 0.8rem 2.4rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #181b1c;
}

.emotion-1::after {
  display: block;
  border-bottom: 3px solid transparent;
  content: '';
  position: absolute;
  bottom: -2px;
  left: 0;
  right: 0;
}

.emotion-1:hover::after {
  border-color: #bec4cd;
}

.emotion-1:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-1::after {
  border-bottom: 3px solid;
}

.emotion-2:focus {
  box-shadow: 0 0 0 0.4rem #f5bbda;
  outline: 0;
}

.emotion-3 {
  color: #3b3f45;
  margin: 0;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

<div>
  <ul
    class="emotion-0"
    role="tablist"
  >
    <li
      role="presentation"
    >
      <a
        aria-selected="true"
        class="emotion-1"
        data-testid="firstTab"
        href="#myTabSet-section-0"
        id="myTabSet-tab-0"
        role="tab"
      >
        About
      </a>
    </li>
  </ul>
  <section
    aria-labelledby="myTabSet-tab-0"
    class="emotion-2"
    data-testid="firstTabSection"
    id="myTabSet-section-0"
    role="tabpanel"
    tabindex="-1"
  >
    <p
      class="Nautilus-Paragraph emotion-3"
    >
      Puppies are cute.
    </p>
  </section>
</div>
`;
